### ISSUE TITLE
Incorrect error message when overloaded function is called with invalid parameter

### ISSUE THREAD

**Issue by bjouhier** [Microsoft/TypeScript#12313]
**TypeScript Version:**  2.0.3

**Code**

```ts
function foo(x: 'a'): void;
function foo(x: 'b'): void;
function foo(x: 'c'): void;
function foo(x: 'a' | 'b' | 'c') {}

foo('d');
```

**Expected behavior:**
```
Error: Argument of type '"d"' is not assignable to parameter of type '"a" | "b" | "c"'
```
**Actual behavior:**
```
Error: Argument of type '"d"' is not assignable to parameter of type '"c"'
```

**Comment by RyanCavanaugh** [comment]
The implementation signature is not part of the type. If this function were in a declaration file you'd just see
```ts
function foo(x: 'a'): void;
function foo(x: 'b'): void;
function foo(x: 'c'): void;
```
It wouldn't be feasible to show a synthesized union type of all the possible first argument types -- some functions have dozens of overloads and that type would be huge (not to mention not even correct)

**Comment by bjouhier** [comment]
Would be nice to have an indication that the parameter has overloads. Something like:

```
Error: Argument of type '"d"' is not assignable to parameter of type '"c"', nor to any of its overloads.
```
